plugins/dashboard/__init__.py

```diff
@@ -684,6 +707,33 @@ def to_dict(self):
         }
 
 
+@execution_cache(ttl=ONE_DAY, key_fn=dataset_key_fn)
```

Comment:
@ritch I think we should cache previews only for the lifetime of the current prompt session. WDYT?

For example, I just did the following:
1. Opened dashboard panel 
2. Clicked `Create plot` and configured a `metadata.size_bytes` numeric histogram
3. Saw that it was empty and so ran `dataset.compute_metadata()` from my shell
4. Reopened the dashboard panel and repeated step 2

Unfortunately, due to the 1 day caching, my preview didn't update in 4.

Reply:
In step 4 you should see the new metadata, since `compute_metadata()` updates the `dataset.last_modified_at` you would get a cache miss.

NOTE: The cached data from step 2 would remain in the ES for 1 day.

Reply:
Ah I didn't inspect the definition of `key_fcn`. That's the problem.

`dataset.last_modified_at` is **not** updated when samples are edited. Sample edits are only detectable via `dataset._max("last_modified_at")`. That's why I'm seeing what I reported in step 4 and not what you're asserting.

The `last_modified_at` field is indexed by default, so inserting this `_max()` call will use an index and thus not require a full collection scan. However, it's still a bit of a heavyweight action to do every time the form updates. Not sure we want to do that.

Conceptually, I would expect to see loading indicators each time I load the edit plot modal.

Reply:
OK I've made the original suggested change. Now the results of the plot loading function will only be cached for the duration of the prompt, and will have a max lifetime of 1hr (to auto-cleanup the results).